### PR TITLE
forgit: fix fish shell, install zsh completions, show sourcing info

### PR DESCRIPTION
**Pull request by cjappl**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

I am one of the maintainers of forgit. There are a few improvements that need to be done to this formula, this is the first round of them.

1. Install zsh completions
2. Fix broken install for fish shell
3. Instruct user on how to source the installed script (This is similar to what something like `autojump` does)

### CODE REVIEW

File: Formula/f/forgit.rb [hunk 1]
```diff
@@ -13,17 +13,32 @@ class Forgit < Formula
...
     pkgshare.install "forgit.plugin.zsh"
     pkgshare.install_symlink "forgit.plugin.zsh" => "forgit.plugin.sh"
   end
 
   def caveats
     <<~EOS
-      A shell plugin has been installed to:
-        #{opt_pkgshare}/forgit.plugin.zsh
-        #{opt_pkgshare}/forgit.plugin.sh
+      A shell plugin has been installed. Please follow these instructions, then restart your shell.
```
Comment (SMillerDev) [inline Formula/f/forgit.rb:29]: I much prefer the previous instructions, it is expected that people know what to do with a plugin

Comment (cjappl) [reply Formula/f/forgit.rb:29]: Before I change this back (which I'm happy to do if you insist), I got this pattern from autojump:

https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/a/autojump.rb#L34-L40

I've run into a problem before when I'm installing something with `brew` that I expect to just do `brew install foo` and instantly be able to use it. I find it useful to have any additional instructions be explicit in the formula, so I can instantly see everything I need to do to get it to work.


Another example:
https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/t/thefuck.rb#L65-L69

Both `autojump` and `thefuck` I have personally messed up before, so these caveats were lifesavers. 



I'd like to state again, more than happy to revert if you insist, just stating my case for keeping it. We can document it on our side if this is unacceptable.


Other examples that are around (basically `grep "source.*HOMEBREW_PREFIX"`
- [zsh-autosuggestions.rb Lines 19-27](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/z/zsh-autosuggestions.rb#L19-L27)
- [zsh-history-substring-search.rb Lines 26-34](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/z/zsh-history-substring-search.rb#L26-L34)
- [zsh-autocomplete.rb Lines 20-29](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/z/zsh-autocomplete.rb#L20-L29)
- [zsh-syntax-highlighting.rb Lines 26-34](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/z/zsh-syntax-highlighting.rb#L26-L34)
- [zsh-autopair.rb Lines 18-26](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/z/zsh-autopair.rb#L18-L26)
- [zsh-navigation-tools.rb Lines 31-38](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/z/zsh-navigation-tools.rb#L31-L38)
- [zsh-you-should-use.rb Lines 16-24](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/z/zsh-you-should-use.rb#L16-L24)
- [powerlevel10k.rb Lines 26-34](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/p/powerlevel10k.rb#L26-L34)
- [root.rb Lines 169-178](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/r/root.rb#L169-L178)
- [antigen.rb Lines 19-30](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/a/antigen.rb#L19-L30)
- [autojump.rb Lines 33-43](https://github.com/Homebrew/homebrew-core/blob/9713251d101b91cf9f96d3a5e2c3331b165b492b/Formula/a/autojump.rb#L33-L43)

Comment (SMillerDev) [reply Formula/f/forgit.rb:29]: The instructions on where to find files are fine, but having an entire manual is not. Caveats should only represent the specific differences between Homebrew and upstream instructions.

Comment (cjappl) [reply Formula/f/forgit.rb]: Understood! Thanks for the feedback, reverted to the original version (plus fish) in this latest crack. 

Let me know if you'd like to see any other changes.

**Review by SMillerDev — APPROVED** [review]

**Pull request merged**